src/units/abilities.cpp

```diff
@@ -1274,6 +1274,24 @@ std::set<std::string> attack_type::alternative_damage_types() const
 	return damage_types;
 }
 
+std::string attack_type::filter_damage_type() const
+{
+	std::pair<std::string, std::string> damages_type = damage_type();
+	if((damages_type.second).empty()){
+		return damages_type.first;
+	}
+
+	unit_ability_list resistance_list;
+	std::string type_damage = damages_type.first;
+	if(other_){
+		resistance_list = (*other_).get_abilities_weapons("resistance", other_loc_, other_attack_, shared_from_this());
+		int first_value = (*other_).resistance_value(resistance_list, damages_type.first);
+		int secondary_value = (*other_).resistance_value(resistance_list, damages_type.second);
+		type_damage = secondary_value > first_value ? damages_type.second : damages_type.first;
+	}
+	return type_damage;
```

Comment:
I assume this is adding another copy of existing logic.

Could filter_damage_type be implemented by calling an existing function?
Could an existing function be simplified by calling filter_damage_type instead (in which case, I'd suggest renaming filter_damage_type to effective_damage_type)?

Reply:
done